sotodlib/preprocess/preprocess_plot.py

```diff
@@ -0,0 +1,239 @@
+matplotlib.use("Agg")
+import matplotlib.pyplot as plt
+from venn import venn
+
+from sotodlib import hwp
+import sotodlib.core as core
+from sotodlib.core.flagman import has_all_cut
+
+def plot_det_bias_flags(aman, det_bias_flags, rfrac_range=(0.1, 0.7),
+                        psat_range=(0, 15), filename="./bias_cuts_venn.png"):
+    """
+    Function for plotting bias cuts.
+
+    Parameters
+    ----------
+    aman : AxisManager
+        Input axis manager.
+    det_bias_flags : AxisManager
+        Output msk_aman of tod_ops.flags.get_det_bias_flags with full_output=True.
+    rfrac_range : Tuple
+        Tuple (lower_bound, upper_bound) for rfrac det selection.
+    psat_range : Tuple
+        Tuple (lower_bound, upper_bound) for P_SAT from IV analysis.
+        P_SAT in the IV analysis is the bias power at 90% Rn in pW.
+    filename : str
+        Full filename with direct path to plot output directory.
+    """
+    msk_names = ['bg', 'r_tes', 'r_frac_gt', 'r_frac_lt', 'p_sat_gt', 'p_sat_lt']
+    msk = det_bias_flags['det_bias_flags']
+    msks = []
+    for name in msk_names:
+        msks.append(det_bias_flags[f'{name}_flags'])
+    all_bad_dets = has_all_cut(msk)
+    msk_ids = []
+    for msk in msks[:2]:
+        bad_dets = has_all_cut(msk)
+        msk_ids.append(np.where(bad_dets == True)[0])
+
+    for i in range(0, 3, 2):
+        bad_dets1 = has_all_cut(msks[i+2])
+        bad_dets2 = has_all_cut(msks[i+3])
+        booleans = [bad_dets1, bad_dets2]
+        union = np.asarray(list(map(any, zip(*booleans))))
+        msk_ids.append(np.where(union == True)[0])
+
+    msk_dict = {'bg < 0': set(msk_ids[0]),
+                'r_tes <= 0': set(msk_ids[1]),
+                f'r_frac < {rfrac_range[0]} or > {rfrac_range[1]}': set(msk_ids[2]),
+                f'p_sat*1e12 < {psat_range[0]} or > {psat_range[1]}': set(msk_ids[3])}
+
+    venn(msk_dict)
+
+    obs_ts = aman.timestamps[0]
+    det = aman.dets.vals[0]
+    plt.title(f"Obs_timestamp:{obs_ts:.0f}\ndet:{det}\nDetectors Cut per Range (Total cut: {len(np.where(all_bad_dets == True)[0])}/{len(aman.dets.vals)})")
+    plt.tight_layout()
+    head_tail = os.path.split(filename)
+    os.makedirs(head_tail[0], exist_ok=True)
+    plt.savefig(filename)
+
+def plot_4f_2f_counts(aman, modes=np.arange(1,49), filename='./4f_2f_counts.png'):
+    """
+    Function for plotting 4f/2f counts for each bandpass.
+
+    Parameters
+    ----------
+    aman : AxisManager
+        Input axis manager.
+    modes : list of int
+        The HWPSS harmonic modes to extract.
+    filename : str
+        Full filename with direct path to plot output directory.
+    """
+    hwpss_ratsatp1 = {}
+    fig, axs = plt.subplots(3, 2, figsize=(15, 15))
+    for i, band in enumerate(['f090', 'f150']):
+        hwpss_ratsatp1[band] = {}
+        m90s = ((aman.det_cal.bg == 0) | (aman.det_cal.bg == 1) | (aman.det_cal.bg == 4) | (aman.det_cal.bg == 5) | (aman.det_cal.bg == 8) | (aman.det_cal.bg == 9))
```

Comment:
Let's not do this...I think if det_info exists we should use `aman.det_info.bandpass == f090` for the 90GHz mask (and similar for 150) and if you want to use `amen.det_cal.bg` then you should use: `m90s = np.isin(man.det_cal.bg, [0,1,4,5,8,9])` for 90 GHz (and similar for 150)